### PR TITLE
feat(test): adding test to verify invalid recipient address over incentivized channel

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #1967

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/fee_middleware_test.go [hunk 1]
```diff
@@ -196,6 +196,122 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_AsyncSingleSender_Succeeds(
...
+	chainARelayerWallet, chainBRelayerWallet, err := s.GetRelayerWallets(relayer)
+	t.Run("relayer wallets fetched", func(t *testing.T) {
+		s.Require().NoError(err)
+	})
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	_, chainBRelayerUser := s.GetRelayerUsers(ctx)
+
+	t.Run("register counter party payee", func(t *testing.T) {
```
Comment (colin-axner) [inline e2e/fee_middleware_test.go:228]:
```suggestion
	t.Run("register counterparty payee", func(t *testing.T) {
```
We reference counterparty as one word throughout the codebase

File: e2e/fee_middleware_test.go [hunk 2]
```diff
@@ -196,6 +196,122 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_AsyncSingleSender_Succeeds(
...
+	t.Run("verify counter party payee", func(t *testing.T) {
+		address, err := s.QueryCounterPartyPayee(ctx, chainB, chainBRelayerWallet.Address, channelA.Counterparty.ChannelID)
+		s.Require().NoError(err)
+		s.Require().Equal(chainARelayerWallet.Address, address)
+	})
+
+	transferAmount := testvalues.DefaultTransferAmount(chainADenom)
+
+	t.Run("send IBC transfer", func(t *testing.T) {
+		transferMsg := transfertypes.NewMsgTransfer(channelA.PortID, channelA.ChannelID, transferAmount, chainAWallet.Bech32Address(chainA.Config().Bech32Prefix), testvalues.InvalidAddress, s.GetTimeoutHeight(ctx, chainA), 0)
```
Comment (colin-axner) [inline e2e/fee_middleware_test.go:243]:
```suggestion
		transferMsg := transfertypes.NewMsgTransfer(channelA.PortID, channelA.ChannelID, transferAmount, chainAWallet.Bech32Address(chainA.Config().Bech32Prefix), testvalues.InvalidAddress, s.GetTimeoutHeight(ctx, chainB), 0)
```

I was a bit worried about this not being clear, but the timeout function returns a valid timeout for the chain referenced. When sending packets, the timeout is for the chain receiving the packet. So if you send from chainA, you want the timeout from chainB

**Review by colin-axner — APPROVED** [review]
Nice work!

File: e2e/fee_middleware_test.go (hunk 2 above)
Comment (chatton) [reply e2e/fee_middleware_test.go:243]: ah I see, thanks for the clarification. I'll make this change.

File: e2e/fee_middleware_test.go (hunk 1 above)
Comment (chatton) [reply e2e/fee_middleware_test.go]: I updated other occurrences of this as well.

File: e2e/fee_middleware_test.go [hunk 3]
```diff
@@ -196,6 +196,122 @@ func (s *FeeMiddlewareTestSuite) TestMsgPayPacketFee_AsyncSingleSender_Succeeds(
...
+			s.Require().NoError(err)
+			s.Require().Len(packets, 1)
+			actualFee := packets[0].PacketFees[0].Fee
+
+			s.Require().True(actualFee.RecvFee.IsEqual(testFee.RecvFee))
+			s.Require().True(actualFee.AckFee.IsEqual(testFee.AckFee))
+			s.Require().True(actualFee.TimeoutFee.IsEqual(testFee.TimeoutFee))
+		})
+
+		t.Run("balance should be lowered by sum of recv ack and timeout", func(t *testing.T) {
```
Comment (crodriguezvega) [inline e2e/fee_middleware_test.go:285]:
```suggestion
		t.Run("balance should be lowered by sum of recv, ack and timeout", func(t *testing.T) {
```

**Review by crodriguezvega — APPROVED** [review]
Good stuff, @chatton!

**Pull request merged**